packages/flutter/lib/src/material/chip.dart

```diff
@@ -10,154 +10,1373 @@ import 'package:flutter/painting.dart';
+///
+///  * [Chip], a chip that displays information and can be deleted.
+///  * [InputChip], a chip represents a complex piece of information, such as an
+///    entity (person, place, or thing) or conversational text, in a compact
+///    form.
+///  * [FilterChip], uses tags or descriptive words as a way to filter content.
+///  * [ActionChip], represents an action related to primary content.
+///  * [CircleAvatar], which shows images or initials of people.
+///  * [Wrap], A widget that displays its children in multiple horizontal or
+///    vertical runs.
+///  * <https://material.google.com/components/chips.html>
+class ChoiceChip extends StatelessWidget
+    implements
+        ChipAttributes, //
+        SelectableChipAttributes,
+        DisabledChipAttributes {
+  const ChoiceChip({
+    Key key,
+    this.avatar,
+    @required this.label,
+    this.labelStyle,
+    this.labelPadding,
+    this.avatarPadding,
+    this.onSelected,
+    @required this.selected,
+    this.selectedColor,
+    this.disabledColor,
+    this.tooltip,
+    this.border: const StadiumBorder(),
+    this.backgroundColor,
+    this.padding,
+  })  : assert(selected != null),
+        assert(label != null),
+        assert(border != null),
+        super(key: key);
+
+  @override
+  final Widget avatar;
+  @override
+  final Widget label;
+  @override
+  final TextStyle labelStyle;
+  @override
+  final EdgeInsetsGeometry labelPadding;
+  @override
+  final EdgeInsetsGeometry avatarPadding;
+  @override
+  final ValueChanged<bool> onSelected;
+  @override
+  final bool selected;
+  @override
+  final Color disabledColor;
+  @override
+  final Color selectedColor;
+  @override
+  final String tooltip;
+  @override
+  final ShapeBorder border;
+  @override
+  final Color backgroundColor;
+  @override
+  final EdgeInsetsGeometry padding;
+
+  @override
+  bool get isEnabled => onSelected != null;
+
+  @override
+  Widget build(BuildContext context) {
+    assert(debugCheckHasMaterial(context));
+    return new RawChip(
+      avatar: avatar,
+      label: label,
+      labelStyle: labelStyle,
+      labelPadding: labelPadding,
+      avatarPadding: avatarPadding,
+      onSelected: onSelected,
+      selected: selected,
+      showCheckmark: false,
+      onDeleted: null,
+      tooltip: tooltip,
+      border: border,
+      disabledColor: disabledColor,
+      selectedColor: selectedColor,
+      backgroundColor: backgroundColor,
+      padding: padding,
+      isEnabled: isEnabled,
+    );
+  }
+}
+
+/// A material design filter chip.
+///
+/// Filter chips use tags or descriptive words as a way to filter content.
+///
+/// Filter chips are a good alternative to [Checkbox] or [Switch] widgets.
+/// Unlike these alternatives, filter chips allow for clearly delineated and
+/// exposed options in a compact area.
+///
+/// Requires one of its ancestors to be a [Material] widget. The [label]
+/// and [border] arguments must not be null.
+///
+/// ## Sample code
```

Comment:
This doesn't look like a FilterChip example

Reply:
That's because it isn't. Fixed.